src/vs/platform/files/node/diskFileSystemProvider.ts

```diff
@@ -172,13 +175,57 @@ export class DiskFileSystemProvider extends AbstractDiskFileSystemProvider imple
 
 	//#region File Reading/Writing
 
-	async readFile(resource: URI): Promise<Uint8Array> {
+	private readonly writeBarriers = new ResourceMap<Barrier>(resource => extUriBiasedIgnorePathCase.getComparisonKey(resource));
```

Comment:
I see you are using `Barrier` for implementation, but it is easier to read and understand if this is called `resourceLocks` is it? I think the purpose is to lock a resource while an operation (read/write) is being happening?